pkgs/applications/system/howdy/pam.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/howdy/src/pam/main.cc b/howdy/src/pam/main.cc
+index 530d5c8..7839a78 100644
+--- a/howdy/src/pam/main.cc
++++ b/howdy/src/pam/main.cc
+@@ -46,7 +46,7 @@ const auto DEFAULT_TIMEOUT =
+     std::chrono::duration<int, std::chrono::milliseconds::period>(100);
+ const auto MAX_RETRIES = 5;
+ const auto PYTHON_EXECUTABLE = "python3";
+-const auto COMPARE_PROCESS_PATH = "/lib/security/howdy/compare.py";
++const auto COMPARE_PROCESS_PATH = "./howdy/compare.py";
```

Comment:
This does not create the proper location for `compare.py`. I thought the `.` would make it append to `$PWD` but it gets built as `//./howdy/compare.py`.

There are two fixes I can think of:
1. Hardcode `/run/current-system/sw/lib/security/howdy/compare.py` in this patch. This may cause problems that I cannot foresee.
2. Build `pam_howdy` in the same `mkDerivation` call as `howdy`, and prepend `$out` to the hardcoded path. Probably the ideal solution.

I will try and go ahead with the second approach.

Reply:
The second approach worked.